addons/udes_sale_stock/models/sale_order.py

```diff
@@ -103,21 +103,51 @@ def get_available_quantity(self, product, locations):
     def cancel_orders_without_availability(self):
         """ From the current list of unconfirmed SO lines, cancel lines that
         cannot be fulfilled with current stock holding """
-        OrderLine = self.env["sale.order.line"]
         Order = self.env["sale.order"]
 
-        _logger.info("Checking orders to cancel due to stock shortage")
-        # Get unreserved stock for each product in locations
-        locations = self.get_available_stock_locations()
-        stock = defaultdict(int)
+        # Get order lines
+        orders = Order.search([("state", "in", ["sale", "draft"])])
+
+        cant_fulfill_lines = orders._find_unfulfillable_order_lines()
```

Comment:
Nit: `unfulfillable_lines` would be a more appropriate variable than `cant_fulfill_lines`. Just a suggestion though so feel free to resolve. :)